Cleanup/experimenter to researcher

okaycj opened the pull request:
Replace all instances of "experimenter" with "researcher".

Closes #559

kimberscott commented on README.md:
```diff
@@ -2,7 +2,7 @@
 For full documentation, see <http://lookit.readthedocs.io> (source: <https://github.com/lookit/lookit-docs/>) 
 
-This is the codebase for the Lookit project, an online developmental laboratory. It includes an Experimenter platform for designing and administering research studies, and a participant-facing platform where users can signup and take part in studies. It is built using Django, PostgreSQL, and Ember.js (see Ember portion of codebase, [ ember-lookit-frameplayer](https://github.com/lookit/ember-looki ...
+This is the codebase for the Lookit project, an online developmental laboratory. It includes a Researcher platform for designing and administering research studies, and a participant-facing platform where users can signup and take part in studies. It is built using Django, PostgreSQL, and Ember.js (see Ember portion of codebase, [ ember-lookit-frameplayer](https://github.com/lookit/ember-lookit-f ...
```
this one can stay since it's referencing the name of the application rather than the users

kimberscott commented on accounts/models.py:
```diff
@@ -78,7 +78,7 @@ class Meta:
             ("can_create_organization", _("Can Create Organization")),
             ("can_remove_organization", _("Can Remove Organization")),
-            ("can_view_experimenter", _("Can View Experimenter")),
+            ("can_view_researcher", _("Can View Researcher")),
```
this we can leave since it's referencing the application ("can view the experimenter app" rather than "can view a particular researcher") - & then won't need migration above

kimberscott commented on accounts/signals.py:
```diff
@@ -1257,8 +1257,8 @@ def post_migrate_create_flatpages(sender, **kwargs):
         			<h3>For researchers</h3>
-        			<p>We are committed to making Lookit a tool for all developmental researchers and to keeping the code open-source. You can view the current <a href="https://github.com/CenterForOpenScience/lookit" target="_blank">Lookit</a>, <a href="https://github.com/CenterForOpenScience/experimenter" target="_blank">Experimenter</a>, and <a href="https://github.com/CenterForOpenScience/exp-addons" t ...
-        			<p>We're not quite ready to start hosting additional studies from other labs yet, but our partners at the Center for Open Science are actively developing functionality for multiple labs to create and post their own studies on Lookit. Although we aim for the platform to be as accessible as possible, some familiarity with programming basics will be helpful for researchers. An ideal skill ...
+        			<p>We are committed to making Lookit a tool for all developmental researchers and to keeping the code open-source. You can view the current <a href="https://github.com/CenterForOpenScience/lookit" target="_blank">Lookit</a>, <a href="https://github.com/CenterForOpenScience/researcher" target="_blank">Researcher</a>, and <a href="https://github.com/CenterForOpenScience/exp-addons" targe ...
+        			<p>We're not quite ready to start hosting additional studies from other labs yet, but our partners at the Center for Open Science are actively developing functionality for multiple labs to create and post their own studies on Lookit. Although we aim for the platform to be as accessible as possible, some familiarity with programming basics will be helpful for researchers. An ideal skill ...
```
these we should leave since they reference the app & repos - it doesn't really matter since these flat pages are just some boilerplate to have something in place

kimberscott commented on exp/templates/exp/_navigation.html:
```diff
@@ -10,7 +10,7 @@
       <ul class="nav navbar-nav navbar-left">
           <li class="nav-item"><a class="navbar-item" href="{% url 'web:home' %}">Lookit</a></li>
-          <li class="nav-item active"><a class="navbar-item active" href="{% url 'exp:study-list' %}">Experimenter</a></li>
+          <li class="nav-item active"><a class="navbar-item active" href="{% url 'exp:study-list' %}">Researcher</a></li>
```
can leave this as the title of the app

kimberscott commented on exp/templates/exp/base.html:
```diff
@@ -7,7 +7,7 @@
     <meta charset="utf-8">
     <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
-    <title>{% block title %}{% endblock %} &mdash; Experimenter</title>
+    <title>{% block title %}{% endblock %} &mdash; Researcher</title>
```
can leave this as title of the app

kimberscott commented on exp/templates/exp/dashboard.html:
```diff
@@ -16,9 +16,9 @@
     <div class="panel-body">
-      <p>You've successfully signed into Experimenter! </p>
+      <p>You've successfully signed into Researcher! </p>
       <p>
-          Your request to join Experimenter has been submitted. You will receive a notification by email when this is completed.
+          Your request to join Researcher has been submitted. You will receive a notification by email when this is completed.
```
can leave these as title of the app

kimberscott commented on exp/urls.py:
```diff
@@ -29,6 +28,7 @@
     ParticipantListView,
     PreviewProxyView,
     RenameVideoView,
+    ResearcherDashboardView,
```
this we *could* leave as ExperimenterDashboardView but it's more ambiguous whether it's the app or the type of user - fine either way

okaycj replied:
I am going to leave it, because the works already done.  Let me know if you change your mind.

kimberscott commented on exp/views/mixins.py:
```diff
@@ -53,19 +53,19 @@ def dispatch(self, request, *args, **kwargs):
-                    "In order to access experimenter pages, you must have two-factor "
+                    "In order to access researcher pages, you must have two-factor "
                     "authentication enabled, and have logged in with your OTP key.",
                 )
                 return redirect("accounts:2fa-login")
         else:
             if user.is_authenticated:
                 return HttpResponseForbidden(
-                    f"Researcher account required to see Experimenter app."
+                    f"Researcher account required to see Researcher app."
                 )
             else:
                 messages.info(
                     request,
-                    "Please sign in with your researcher account to see Experimenter app.",
+                    "Please sign in with your researcher account to see Researcher app.",
```
leave these as Experimenter since referencing the app

kimberscott commented on locale/de/LC_MESSAGES/django.po:
```diff
@@ -262,7 +262,7 @@ msgid "Can Remove Organization"
 
 #: accounts/models.py:70
-msgid "Can View Experimenter"
+msgid "Can View Researcher"
 msgstr "Kann den Experimentator sehen"
 
```
(the translations should end up being unnecessary if we don't make the changes to the template text / permission names as noted above)

kimberscott commented on project/settings.py:
```diff
@@ -126,7 +126,7 @@
 ROOT_URLCONF = "project.urls"
 LOGIN_URL = "/login/"
-EXPERIMENTER_LOGIN_URL = "/accounts/login/"
+RESEARCHER_LOGIN_URL = "/accounts/login/"
```
I think we can leave this since referncing the app - it also makes me a little nervous that we don't change this somewhere else :)

kimberscott commented on studies/templates/emails/download_framedata_dict.html:
```diff
@@ -8,5 +8,5 @@
   <br/>
   <br/>
-  <em>For security reasons this link will only work for 1 day. If you need to access this archive after 1 day you will have to rebuild it in experimenter.</em>
+  <em>For security reasons this link will only work for 1 day. If you need to access this archive after 1 day you will have to rebuild it in researcher.</em>
```
can leave this since referencing app

kimberscott commented on studies/templates/emails/download_framedata_dict.txt:
```diff
@@ -4,4 +4,4 @@ Your frame data dictionary has been created and is ready for download.
 
-For security reasons this link will only work for 1 day. If you need to access this archive after 1 day you will have to rebuild it in experimenter.
\ No newline at end of file
+For security reasons this link will only work for 1 day. If you need to access this archive after 1 day you will have to rebuild it in researcher.
```
can leave this since referencing app

kimberscott commented on studies/templates/emails/download_zip.html:
```diff
@@ -8,5 +8,5 @@
   <br/>
   <br/>
-  <em>For security reasons this link will only work for 30m. If you need to access this archive after 30m you will have to rebuild it in experimenter.</em>
+  <em>For security reasons this link will only work for 30m. If you need to access this archive after 30m you will have to rebuild it in researcher.</em>
```
can leave this since referencing app

kimberscott commented on studies/templates/emails/download_zip.txt:
```diff
@@ -4,4 +4,4 @@ Your video archive has been created and is ready for download.
 {{signed_url|safe}}
 
-For security reasons this link will only work for 30m. If you need to access this archive after 30m you will have to rebuild it in experimenter.
+For security reasons this link will only work for 30m. If you need to access this archive after 30m you will have to rebuild it in researcher.
```
can leave this since referencing app

kimberscott commented on web/templates/web/_navigation.html:
```diff
@@ -14,7 +14,7 @@
         <ul class="nav navbar-nav navbar-left">
           <li class="nav-item active"><a class="navbar-item" href="{% url 'web:home' %}">{% trans "Lookit" %}</a></li>
-          <li class="nav-item"><a class="navbar-item active" href="{% url 'exp:study-list' %}">{% trans "Experimenter" %}</a></li>
+          <li class="nav-item"><a class="navbar-item active" href="{% url 'exp:study-list' %}">{% trans "Researcher" %}</a></li>
```
can leave this since referencing app

kimberscott reviewed (changes requested):
We can leave "Experimenter" in a bunch of places where we're referencing the name of the application rather than type of user (which will also avoid changing translation strings), otherwise looks good!

okaycj commented:
> We can leave "Experimenter" in a bunch of places where we're referencing the name of the application rather than type of user (which will also avoid changing translation strings), otherwise looks good!

Thank you @kimberscott!  I have removed the migration.  Additionally, I have fixed an merge error in the new views (https://github.com/lookit/lookit-api/pull/685/commits/a1bea914f4565a684fe6eedea236f7b07a24378c) and updated per your above instructions (https://github.com/lookit/lookit-api/pull/685/commits/13ca9e02c465cc7bb91a3d3a9ed165e8e4fcc90b).  Let me know if you find any more or if I've made a mistake with these changes.

kimberscott commented on accounts/signals.py:
```diff
@@ -1257,8 +1257,8 @@ def post_migrate_create_flatpages(sender, **kwargs):
         			<h3>For researchers</h3>
-        			<p>We are committed to making Lookit a tool for all developmental researchers and to keeping the code open-source. You can view the current <a href="https://github.com/CenterForOpenScience/lookit" target="_blank">Lookit</a>, <a href="https://github.com/CenterForOpenScience/experimenter" target="_blank">Experimenter</a>, and <a href="https://github.com/CenterForOpenScience/exp-addons" t ...
-        			<p>We're not quite ready to start hosting additional studies from other labs yet, but our partners at the Center for Open Science are actively developing functionality for multiple labs to create and post their own studies on Lookit. Although we aim for the platform to be as accessible as possible, some familiarity with programming basics will be helpful for researchers. An ideal skill ...
+        			<p>We are committed to making Lookit a tool for all developmental experimenters and to keeping the code open-source. You can view the current <a href="https://github.com/CenterForOpenScience/lookit" target="_blank">Lookit</a>, <a href="https://github.com/CenterForOpenScience/experimenter" target="_blank">Experimenter</a>, and <a href="https://github.com/CenterForOpenScience/exp-addons" ...
+        			<p>We're not quite ready to start hosting additional studies from other labs yet, but our partners at the Center for Open Science are actively developing functionality for multiple labs to create and post their own studies on Lookit. Although we aim for the platform to be as accessible as possible, some familiarity with programming basics will be helpful for experimenter. An ideal skil ...
```
just for clarity can this stay as it was?

okaycj replied:
Actually, I think I understand what you mean.  I'll put up a commit and take my chances.

kimberscott commented on exp/templates/exp/dashboard.html:
```diff
@@ -16,7 +16,7 @@
     </div>
     <div class="panel-body">
-      <p>You've successfully signed into Experimenter! </p>
+      <p>You've successfully signed into Researcher! </p>
```
keep as Experimenter (referring to app) although to be fair I think this is unused

kimberscott reviewed (changes requested):
Sorry for the multiple rounds on this in-principle-tiny change - a few minor places to keep Experimenter noted, and then we should just revert the changes to the translation files

kimberscott reviewed (approved):
Looks great! This one was a bit confusing with Experimenter being used for both an app & a user type; one more piece of odd history down :)
